product-service/src/main/java/org/arpha/exception/ProductExceptionHandler.java

```diff
@@ -0,0 +1,30 @@
+package org.arpha.exception;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+@Slf4j
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class ProductExceptionHandler {
+
+    @ExceptionHandler({CreateCategoryException.class, CreateGenreException.class, CreateProductException.class, DeleteEntityException.class})
+    public ProblemDetail handleProductBadRequest(RuntimeException ex) {
+        log.error(StringUtils.EMPTY, ex);
+        return ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, ex.getMessage());
+    }
```

Comment:
Can you please create some generic exception and handle it?. Lets try to avoid injecting RuntimeException

Reply:
Why is that bad?